zombies.js

```diff
@@ -1,3 +1,162 @@
+// jshint esversion: 6
+class Item {
+  constructor(name) {
+    this.name = name;
+  }
+}
+
+class Weapon extends Item {
+  constructor(name, damage) {
+    super(name);
+    this.damage = damage;
+  }
+}
+
+class Food extends Item {  
+  constructor(name, energy) {
+    super(name);
+    this.energy = energy;
+  }
+}
+
+class Player {
+  constructor(name, health, strength, speed) {
+    this._pack = [];
+    this._maxHealth = health;
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this.isAlive = true;
+    this.equipped = false;
+  }
+
+  getPack() {
+    return this._pack;
+  }
+
+  getMaxHealth() {
+    return this._maxHealth;
+  }
+
+  checkPack() {
+    let playerPack = this._pack;
+    console.log('Contents: ');
+    for (let i in playerPack.length) {
+      console.log(playerPack[i] + ", ");
+    }
+  }
+
+  takeItem(item) {
+    let packContents = this._pack;
+    if (packContents.length > 2 && !packContents.includes(item)) {
+      console.log('Your pack is full.');
+      return false;
+    } else {
+      packContents.push(item);
+      console.log(this.name + ' picked up ' + item.name);
+      return true;
+    }
+  }
+
+  discardItem(item) {
+    let itemIndex = this._pack.indexOf(item);
+
+    if (itemIndex !== -1) {
+      this._pack.splice(itemIndex, 1);
+      console.log(this.name + ' ' + item.name + ' was discarded.');
+      return true;
+    } else {
+      console.log(item.name + ' nothing was discarded since the item could not be found.');
+      return false;
+    }
+  }
+
+  equip(itemToEquip) {
+    let itemIndex = this._pack.indexOf(itemToEquip);
+    let equipped = this.equipped;
+    let _pack = this._pack;
+
+    if (!equipped && _pack.includes(itemToEquip) && itemToEquip instanceof Weapon) {
+      this.equipped = itemToEquip;
+      this._pack.splice(itemIndex, 1);
+    } else if (equipped && _pack.includes(itemToEquip) && itemToEquip instanceof Weapon) {
+      let tempItem = this.equipped;
+      this.equipped = itemToEquip;
+      this._pack.splice(itemIndex, 1);
+      this._pack.push(tempItem);
+    }
+  }
+
+  eat(itemToEat) {
+    let energyCount = itemToEat.energy;
+
+    if (this._pack.includes(itemToEat) && itemToEat instanceof Food) {
+      this._pack.splice(this._pack.indexOf(itemToEat), 1);
+      this.health = this.health + energyCount;
+
+      if (this.health > this.getMaxHealth()) {
+        this.health = this._maxHealth;
+      }
+    }
+  }
+
+  useItem(item) {
+    if (item instanceof Weapon) {
+      this.equip(item);
+    } else if (item instanceof Food) {
+      this.eat(item);
+    }
+  }
+
+  equippedWith() {
+    if (!this.equipped) {
```

Comment:
You don't need your false case to always be first. If you are going to handle both true and false with if else, than just make if the truthy logic.
```javascript
if(this.equipped) {
  console.log(this.name + ' has ' + this.equipped + ' equipped.');
  return this.equipped.name;
} else {
  console.log('No weapon equipped');
  return false;
}